lib/uri_parser.js

```diff
@@ -225,6 +227,7 @@ function applyConnectionStringOption(obj, key, value, options) {
   } else if (key === 'appname') {
     value = decodeURIComponent(value);
   } else if (key === 'readconcernlevel') {
+    obj['readConcernLevel'] = value;
```

Comment:
The spec tests check that `readConcernLevel` is set instead of `readConcern: { level: x }`.